From f44b6b85619276812556b794c3b546d5bb7b3761 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:36 -0400
Subject: [net] igmp: Call skb_checksum_simple_validate

Message-id: <1409652920-431-14-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90142
O-Subject: [PATCH 13/57] [RHEL7.1 net] igmp: Call skb_checksum_simple_validate
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit de08dc1a8e7052f4cbaf920ce0af6bb261595705
Author: Tom Herbert <therbert@google.com>
Date:   Wed May 7 16:52:10 2014 -0700

    igmp: Call skb_checksum_simple_validate

    Use skb_checksum_simple_validate to verify checksum.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/igmp.c b/net/ipv4/igmp.c
index 9feae66..65c6498 100644
--- a/net/ipv4/igmp.c
+++ b/net/ipv4/igmp.c
@@ -962,16 +962,8 @@ int igmp_rcv(struct sk_buff *skb)
  if (!pskb_may_pull(skb, sizeof(struct igmphdr)))
   goto drop;
 
- switch (skb->ip_summed) {
- case CHECKSUM_COMPLETE:
-  if (!csum_fold(skb->csum))
-   break;
-  /* fall through */
- case CHECKSUM_NONE:
-  skb->csum = 0;
-  if (__skb_checksum_complete(skb))
-   goto drop;
- }
+ if (skb_checksum_simple_validate(skb))
+  goto drop;
 
  ih = igmp_hdr(skb);
  switch (ih->type) {
-- 
1.7.1